### PR TITLE
Compaction fix

### PR DESCRIPTION
**Pull request by adam-yorwerth**
Small changes:
- Compaction threshold considered when looking for the unique set of keys now, rather than all keys as before. Changes in both Postgres and sqlite for this.
- Tests changed to reflect this for both storage implementations
- There was a lack of tests in Postgres around compaction, so some have been added.
- Small refactoring of queries.

### CODE REVIEW

File: pipe-storage-postgresql/src/integration/groovy/com/tesco/aqueduct/pipe/storage/PostgresqlStorageIntegrationSpec.groovy [hunk 1]
```diff
@@ -36,7 +38,10 @@ class PostgresqlStorageIntegrationSpec extends StorageSpec {
         sql = new Sql(pg.embeddedPostgres.postgresDatabase.connection)
 
         dataSource = Mock()
-        dataSource.connection >> pg.embeddedPostgres.postgresDatabase.connection
+
+        dataSource.connection >> {
+            DriverManager.getConnection(pg.embeddedPostgres.getJdbcUrl("postgres", "postgres"))
+        }
```
Comment (adam-yorwerth) [inline pipe-storage-postgresql/src/integration/groovy/com/tesco/aqueduct/pipe/storage/PostgresqlStorageIntegrationSpec.groovy]: This allows multiple connections during the test

File: pipe-storage-postgresql/src/main/java/com/tesco/aqueduct/pipe/storage/PostgresqlStorage.java [hunk 2]
```diff
@@ -197,17 +199,19 @@ private static String getCompactionQuery() {
...
+            "   FROM " +
             "   events e, " +
             "   ( " +
-            "     select msg_key, max(msg_offset) max_offset " +
-            "     from events " +
-            "     group by msg_key " +
+            "     SELECT msg_key, max(msg_offset) max_offset " +
+            "     FROM events " +
+            "     WHERE " +
+            "       created_utc <= ? " +
```
Comment (adam-yorwerth) [inline pipe-storage-postgresql/src/main/java/com/tesco/aqueduct/pipe/storage/PostgresqlStorage.java]: github hides it, this is the actual non-formatting change here

File: pipe-storage-postgresql/src/integration/groovy/com/tesco/aqueduct/pipe/storage/PostgresqlStorageIntegrationSpec.groovy [hunk 3]
```diff
@@ -182,9 +187,73 @@ class PostgresqlStorageIntegrationSpec extends StorageSpec {
         result.messages.isEmpty()
     }
 
+    def 'All duplicate messages are compacted for whole data store'() {
```
Comment (cwebbtw) [inline pipe-storage-postgresql/src/integration/groovy/com/tesco/aqueduct/pipe/storage/PostgresqlStorageIntegrationSpec.groovy]: These tests read well 👍

File: pipe-storage-postgresql/src/main/java/com/tesco/aqueduct/pipe/storage/PostgresqlStorage.java (hunk 2 above)
Comment (cwebbtw) [reply pipe-storage-postgresql/src/main/java/com/tesco/aqueduct/pipe/storage/PostgresqlStorage.java]: Looks better

**Pull request merged**